### PR TITLE
5 implement use case 5 edit event

### LINKED ISSUE #17: Implement Use Case #5 - Edit Event

### PR DESCRIPTION
**Pull request by ssfaron**
Closes #17 #19

### CODE REVIEW

File: source_code/backend/CS_department_monitor_backend/src/main/java/com/cs5324/monitorbackend/service/EventService.java [hunk 1]
```diff
@@ -1,15 +1,67 @@
...
+    public Notification submitEdits(Event editedEvent){
+        editedEvent.setApprovalStatus(ItemStatus.PENDING);
+        editedEvent.setDateOfEvent(editedEvent.getDateOfEvent());
+        editedEvent.setPage(editedEvent.getPage());
+
+        Notification editedNotif = new Notification();
+        editedEvent.setNotification(editedNotif);
+
+        return editedNotif;
+    }
```
Comment (Matt-Hays) [inline source_code/backend/CS_department_monitor_backend/src/main/java/com/cs5324/monitorbackend/service/EventService.java]: - What if the given event, passed in from the controller, does not have a valid id?
- What if either `editedEvent.dateOfEvent == null` or `editedEvent.page == null`, will we overwrite the existing data with `null`?
- The edited event should be saved to the database?

**Review by Matt-Hays — CHANGES_REQUESTED** [review]
See review comments.

File: source_code/backend/CS_department_monitor_backend/src/main/java/com/cs5324/monitorbackend/service/EventService.java [hunk 2]
```diff
@@ -1,15 +1,67 @@
...
+
 @Slf4j
 @Service
 @RequiredArgsConstructor
 public class EventService{
     @Resource
     private final EventRepository eventRepo;
+
+    @Resource
+    private final UserRepository userRepo;
```
Comment (tkm3d1a) [inline source_code/backend/CS_department_monitor_backend/src/main/java/com/cs5324/monitorbackend/service/EventService.java]: We should swap this to utilize the UserService class and not point directly to the repo

File: source_code/backend/CS_department_monitor_backend/src/main/java/com/cs5324/monitorbackend/service/EventService.java [hunk 3]
```diff
@@ -1,15 +1,67 @@
...
+
+    public Set<Event> getEvents(UUID userId){
+
+        Optional<User> user = userRepo.findById(userId);
```
Comment (tkm3d1a) [inline source_code/backend/CS_department_monitor_backend/src/main/java/com/cs5324/monitorbackend/service/EventService.java]: UserService should handle dealing with Optional<User> objects

**Review by tkm3d1a — CHANGES_REQUESTED** [review]
Remove UserRepo usage and only use UserService when fetching users - other than that good to go

**Review by tkm3d1a — APPROVED** [review]

**Review by Matt-Hays — APPROVED** [review]

**Pull request merged**